src/util/prependDeep.ts

```diff
@@ -22,8 +22,8 @@ export default function merge(
         target[key] = sourceVal;
       } else if (targetVal === sourceVal) {
         continue;
-      } else if (isSchema(targetVal)) {
-        if (isSchema(sourceVal)) target[key] = sourceVal.concat(targetVal);
+      } else if (targetVal instanceof MixedSchema) {
```

Comment:
what specifically is this fixing? Likely the more correct fix is to expand the Schema interface such that it is accurate. This may generally require a lot of typing rework. There are lots of places that use MixedSchema simply as a shortcut/laziness on my part while trying to verify the type approach would work

Reply:
I was trying to solve the typing issues that Typescript reported. 

Not all schema types have concat I think. Other solution might be changing Schema from interface to class that has a concat method and use `instanceof` to detect if a given value is a schema at all

Reply:
yeah fair, this is ultimately a fairly niche-y case